app/models/shopify_client.rb

```diff
@@ -0,0 +1,58 @@
+class ShopifyClient
+  attr_reader :oauth_uid
+  attr_reader :oauth_token
+
+  def initialize(oauth_uid, oauth_token)
+    @oauth_uid = oauth_uid
+    @oauth_token = oauth_token
+  end
+
+
+  def find(external_entity_name, params = {})
+    with_shopify_session do
+      # ugly hack, did not find a better way to convert an entity to a  seriable hash (serializable_hash was not recursive)
+      find_all(get_shopify_entity_constant(external_entity_name), params).map { |x| JSON.parse(x.to_json) }
```

Comment:
unfortunately, the find_all may return a Shopify object instance, that can contains nested other Shopify object.